File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoCodigoBarras.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.Commands;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand;
import com.google.gson.JsonObject;

public class ImpressaoCodigoBarras extends TermicaCommand {
    final private int tipo;
    final private String dados;
    final private int altura;
    final private int largura;
    final private int HRI;

    public ImpressaoCodigoBarras(int tipo, String dados, int altura, int largura, int hri) {
        super("ImpressaoCodigoBarras");
        this.tipo = tipo;
        this.dados = dados;
        this.altura = altura;
        this.largura = largura;
        HRI = hri;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("tipo", this.tipo);
        functionParametersJson.addProperty("dados", this.dados);
        functionParametersJson.addProperty("altura", this.altura);
        functionParametersJson.addProperty("largura", this.largura);
        functionParametersJson.addProperty("HRI", this.HRI);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoCodigoBarras.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand
import com.google.gson.JsonObject


class ImpressaoCodigoBarras(
    private val tipo: Int,
    private val dados: String,
    private val altura: Int,
    private val largura: Int,
    private val HRI: Int
) :
    TermicaCommand("ImpressaoCodigoBarras") {
    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("tipo", tipo)
        functionParametersJson.addProperty("dados", dados)
        functionParametersJson.addProperty("altura", altura)
        functionParametersJson.addProperty("largura", largura)
        functionParametersJson.addProperty("HRI", HRI)
        return functionParametersJson
    }
}